File: program/find-perimeter-of-a-square/FindPerimeterOfASquare.c
#include <stdio.h>
#include <setjmp.h> // For Exception Handling in C

jmp_buf savebuff;

#define TRY if (setjmp(savebuff) == 0)
#define CATCH else

// Declaring a function with exception handling
float calc_peri(float side);

// Main function
int main()
{
    float s;                                                  // Declaring Variable
    printf("Enter the side of square to find perimeter: \n"); // Input prompt
    scanf("%f", &s);                                          // Reading input

    TRY
    {
        printf("Perimeter of square is: %f\n", calc_peri(s));
    }
    CATCH
    {
        printf("ERROR!!! invalid input.\n");
    }

    return 0;
}

float calc_peri(float side)
{
    float peri;
    peri = 4 * side;
    if (side < 0)
    {
        longjmp(savebuff, 1);
    }

    return peri;
}

File: program/find-perimeter-of-a-square/FindPerimeterOfASquare.java
import java.util.Scanner;
public class FindPerimeterOfASquare
{
	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter the side of the square:");
		double peri = sc.nextDouble();
		double  perimeter = 4*peri;
		System.out.println("Perimeter: " + perimeter);
	}
}

// Contributed by Raghav Garg (@grraghav120)
